flocker/acceptance/testtools.py

```diff
@@ -936,9 +936,21 @@ def nodes_available():
         ).write()
 
         def failed_query(failure):
-            Message.new(message_type="acceptance:is_available_error",
-                        reason=unicode(failure),
-                        exception=unicode(failure.__class__)).write()
+            try:
+                # Try to detect _WrapperException.
+                reasons = failure.value.reasons
+            except AttributeError:
```

Comment:
I'd rather have `getattr(failure.value, "reasons", None) is None` or even `isinstance`.